### PR TITLE
rework reponse processing to handle element subtypes

### PR DESCRIPTION
**Pull request by danrubel**
@devoncarew 
+ rework reponse processing to handle element subtypes
+ simplify GetLibraryConsumer and GetInstanceConsumer
+ rename ObservatoryConst to VmServiceConst

### CODE REVIEW

File: dart/tool/java/generate_java.dart [hunk 1]
```diff
@@ -407,9 +398,17 @@ class Method extends Member {
...
+    types.sort((t1, t2) => t1.name.compareTo(t2.name));
     for (var t in types) {
+
+      // TODO(danrubel) rename classes to prevent collision
+      // with common java types. e.g. "Class" --> "DartClass"
+      if (t.name == 'Class' || t.name == 'Error') continue;
+
       var jsonType = t.name;
       var responseName = classNameFor(t.elementTypeName);
       writer.addLine('  if (responseType.equals("$jsonType")) {');
```
Comment (devoncarew) [inline dart/tool/java/generate_java.dart]: The doc writer class does have some methods which handle indenting and unindenting automatically, so the client doesn't have to explicitly send in indent strings. FYI for future changes :)

**Comment by devoncarew** [comment]
lgtm

File: dart/tool/java/generate_java.dart (hunk 1 above)
Comment (danrubel) [inline dart/tool/java/generate_java.dart]: Good to know. Thanks!

**Comment by danrubel** [comment]
Merged as aee2f99e581e1f261033084f2fb9ad56e313b30c

**Pull request closed without merging**